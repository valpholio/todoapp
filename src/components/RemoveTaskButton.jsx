import React from 'react';
import PropTypes from 'prop-types';

const RemoveTaskButton = ({removeTask = () => {}, id}) => {
    return (
            <button onClick={() => removeTask(id)} className="btn btn-default btn-sm">
                <span className="glyphicon glyphicon-remove"></span> Remove
            </button>
    );
};

RemoveTaskButton.propTypes = {
    id: PropTypes.any.isRequired,
    removeTask: PropTypes.func
};

export default RemoveTaskButton;