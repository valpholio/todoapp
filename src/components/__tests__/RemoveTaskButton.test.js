import React from 'react';
import renderer from 'react-test-renderer';
import RemoveTaskButton from '../RemoveTaskButton.jsx';

describe('RemoveTaskButton', () => {
    test('snapshot renders', () => {
        const component = renderer.create(<RemoveTaskButton id='1'/>);
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
