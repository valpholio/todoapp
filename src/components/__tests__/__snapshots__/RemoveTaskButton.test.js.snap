// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveTaskButton snapshot renders 1`] = `
<button
  className="btn btn-default btn-sm"
  onClick={[Function]}
>
  <span
    className="glyphicon glyphicon-remove"
  />
   Remove
</button>
`;
